src/sas/sasgui/perspectives/fitting/fitpage.py

```diff
@@ -1238,7 +1238,12 @@ def _on_select_model(self, event=None):
             custom_model = CUSTOM_MODEL
             mod_cat = self.categorybox.GetStringSelection()
             if mod_cat == custom_model:
+                ################This is where the model dies###############
+                temp_id = self.model.id
                 temp = self.parent.update_model_list()
+                for v in self.parent.model_dictionary.values():
```

Comment:
The size of model_dictionary should be small enough and this function called rarely enough that the delay from generating the value list should be imperceptible.

It's only adding a single line of code, so I'm not opposed to it if anyone has a strong opinion on it, but I find the current version marginally more legible.